src/schemathesis/_hypothesis.py

```diff
@@ -107,38 +143,77 @@ def is_surrogate(item: Any) -> bool:
     """
+    if not isinstance(query, dict):
+        return False
     for name, value in query.items():
         if is_surrogate(name) or is_surrogate(value):
             return False
     return True
 
 
-def get_case_strategy(endpoint: Endpoint) -> st.SearchStrategy:
+def is_valid_cookie(cookies: Any) -> bool:
+    if not isinstance(cookies, dict):
+        return False
+    for name, value in cookies.items():
+        if not isinstance(name, str) or not isinstance(value, str):
+            return False
+        if not utils.is_ascii_encodable(name) or not utils.is_ascii_encodable(value):
+            return False
+    return True
+
+
+def is_valid_form_data(form_data: Any) -> bool:
+    if isinstance(form_data, Mapping):
+        return all(
+            isinstance(key, str) and utils.is_ascii_encodable(key) and isinstance(value, (bytes, str, int))
+            for key, value in form_data.items()
+        )
+
+    def is_valid_item(item: Any) -> bool:
+        return (
+            isinstance(item, Sequence)
+            and len(item) == 2
+            and isinstance(item[0], str)
+            and isinstance(item[1], (bytes, str, int))
+        )
+
+    return isinstance(form_data, (tuple, list, set)) and all(is_valid_item(item) for item in form_data)
+
+
+def get_schema_strategy(value: Any, input_type: InputType) -> st.SearchStrategy:
+    strategy_func = {InputType.valid: from_schema, InputType.invalid: not_from_schema}[input_type]
+    return strategy_func(value)
+
+
+def get_case_strategy(endpoint: Endpoint, input_type: InputType = InputType.valid) -> st.SearchStrategy:
     """Create a strategy for a complete test case.
 
     Path & endpoint are static, the others are JSON schemas.
     """
     strategies = {}
-    static_kwargs: Dict[str, Any] = {"endpoint": endpoint}
+    static_kwargs: Dict[str, Any] = {"endpoint": endpoint, "input_type": input_type}
     try:
         for parameter in PARAMETERS:
             value = getattr(endpoint, parameter)
             if value is not None:
+                strategy = get_schema_strategy(value, input_type)
                 if parameter == "path_parameters":
-                    strategies[parameter] = (
-                        from_schema(value).filter(filter_path_parameters).map(quote_all)  # type: ignore
-                    )
+                    strategies[parameter] = strategy.filter(filter_path_parameters).map(quote_all)  # type: ignore
                 elif parameter in ("headers", "cookies"):
-                    strategies[parameter] = from_schema(value).filter(is_valid_header)  # type: ignore
+                    strategies[parameter] = strategy.filter(is_valid_header)  # type: ignore
                 elif parameter == "query":
-                    strategies[parameter] = from_schema(value).filter(is_valid_query)  # type: ignore
+                    strategies[parameter] = strategy.filter(is_valid_query)  # type: ignore
+                elif parameter == "cookies":
```

Comment:
Do we want this block to execute? The `elif parameter in ("headers", "cookies")` will evaluate to True before we reach `elif parameter == "cookies"`.

Reply:
Indeed! Will clean it up :) Thanks

At the moment I am not sure when I can get back to this pull request :(

Reply:
That's OK! I'm just sifting through this to review as part of reviewing how requests are generated. I'm going to start opening some issues for features (one of them will be modify requests before they are sent), and I will start taking issues off the board in about a week.